cmake/external/Cereal.cmake

```diff
@@ -0,0 +1,16 @@
+GetCompilers()
+
+include(ExternalProject)
+ExternalProject_Add(
+    Cereal
+    PREFIX ${CMAKE_SOURCE_DIR}/Cereal
+    DOWNLOAD_DIR ${THIRD_PARTY_DIR}/Cereal
+    URL https://github.com/USCiLab/cereal/archive/v1.1.2.tar.gz
+    URL_MD5 34d4ad174acbff005c36d4d10e48cbb9
+	)
+
+SET(CEREAL_INCLUDE_DIRS ${THIRD_PARTY_DIR}/include/cereal)
```

Comment:
so this should work
SET(CEREAL_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/Cereal/src/Cereal/include)